sdk/core/System.Net.ClientModel/src/ModelReaderWriter/IModel.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace System.Net.ClientModel.Core
+{
+    /// <summary>
+    /// Allows an object to control its own writing and reading.
+    /// The format is determined by the implementer.
+    /// </summary>
+    /// <typeparam name="T">The type the model can be converted into.</typeparam>
+    public interface IModel<out T>
```

Comment:
I defer to @KrzysztofCwalina on where we want to land with names prior to the review with the BCL team and feedback we might get with them, but I wanted to document some observations I have (non-blocking for this PR):
- Could the `IModel` name be confused with the package name `ClientModel`?  
  - Questions on this might include: Why are we modifying the name `Model` in the package name but not in this interface name?
- What does the word "model" mean in BCL APIs like `ClientModel` and `ServiceModel` and elsewhere, and is this use of the word model consistent with that vocabulary and meaning?
- Where we also have IJsonModel - is the model itself about Json?  Why is `Json` the best modifier for `Model`?

One alternative proposal I might toss out there as a starting point for conversation -- and motivated by the observation that .NET interfaces are often used to describe the behaviors enable (per .NET not supporting multiple inheritance), i.e. IEnumerable and IDisposable vs. the "is-a" style interfaces like IList and ICollection -- would be what about something like `IContentReadWriteable` and `IJsonContentReadWriteable`?

Reply:
One additional comment here is that the "is-a" interfaces like IList and ICollection have a lot of APIs that are required for those concrete types.  Adding a method to an interface after it GAs is a breaking change so by calling an interface with only `Read` and `Write` operations `IModel` sort of means "cloud client models only read and write their content".  Is there a possible future world where a cloud client model would have other behaviors we wanted to enforce via interfaces?

Reply:
One other angle on this as a BCL type might be ... would any other non-model component ever want to use these behaviors as specified in the interface, and so we would want to name the interface to describe the behavior so that that would make sense?